File: backend/models/transaction.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from datetime import datetime

class TransactionData(BaseModel):
    model_config = ConfigDict(extra='allow')
    
    date: str
    description: str
    withdrawal: Optional[int] = None
    deposit: Optional[int] = None
    balance: int
    confidence_score: float = 0.0

class ProcessingResult(BaseModel):
    transactions: List[TransactionData]
    confidence_score: float
    processing_method: str
    claude_confidence: Optional[float] = None
    gpt4v_confidence: Optional[float] = None
    agreement_score: Optional[float] = None

File: frontend/src/types/transaction.ts
export interface TransactionData {
  date: string;
  description: string;
  withdrawal?: number;
  deposit?: number;
  balance: number;
  confidence_score: number;
  [key: string]: any; // 動的な列追加に対応
}

export interface ProcessingResult {
  id: string;
  filename: string;
  status: string;
  transactions: TransactionData[];
  confidence_score: number;
  processing_time: number;
  processing_method?: string;
  claude_confidence?: number;
  gpt4v_confidence?: number;
  agreement_score?: number;
}